src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2944,17 +2946,37 @@ def openshift_create(cmd, client, resource_group_name, name,  # pylint: disable=
                 namespace='Microsoft.Network', type='virtualNetwork',
                 name=vnet_peer
             )
+    if loganalytics_workspace_resource_id is not None:
+        loganalytics_workspace_resource_id = loganalytics_workspace_resource_id.strip()
+        if not loganalytics_workspace_resource_id.startswith('/'):
+            loganalytics_workspace_resource_id = '/' + loganalytics_workspace_resource_id
+        if loganalytics_workspace_resource_id.endswith('/'):
+            loganalytics_workspace_resource_id = loganalytics_workspace_resource_id.rstrip('/')
+        monitor_profile = OpenShiftManagedClusterMonitorProfile(enabled=True, loganalytics_workspace_resource_id=loganalytics_workspace_resource_id)
+    else:
+        monitor_profile = None
 
     network_profile = NetworkProfile(vnet_cidr=vnet_prefix, peer_vnet_id=vnet_peer)
 
-    osamc = OpenShiftManagedCluster(
-        location=location, tags=tags,
-        open_shift_version="v3.11",
-        network_profile=network_profile,
-        auth_profile=auth_profile,
-        agent_pool_profiles=agent_pool_profiles,
-        master_pool_profile=agent_master_pool_profile,
-        router_profiles=[default_router_profile])
+    if monitor_profile is not None:
+        osamc = OpenShiftManagedCluster(
```

Comment:
Looking above...

```python
from azure.mgmt.containerservice.v2019_04_30.models import OpenShiftManagedCluster
```

These are both referring to the `v2019_04_30` model, we need to import the `v2019_09_30_preview` for the Azure Monitor case.

Reply:
😬 How did I miss that 🤦‍♂ ?  Updated with an alias.